client.go

```diff
@@ -11,15 +11,17 @@ import (
 	"golang.org/x/oauth2"
 )
 
-// Endpoint is the OAuth2 endpoint for authenticating with the Tesla API.
-var Endpoint = oauth2.Endpoint{
-	AuthURL:  "https://auth.tesla.com/oauth2/v3/authorize",
-	TokenURL: "https://auth.tesla.com/oauth2/v3/token",
+// DefaultOAuth2Config is the OAuth2 configuration for authenticating with the Tesla API.
+var DefaultOAuth2Config = &oauth2.Config{
+	ClientID:    "ownerapi",
+	RedirectURL: "https://auth.tesla.com/void/callback",
+	Endpoint: oauth2.Endpoint{
+		AuthURL:  "https://auth.tesla.com/oauth2/v3/authorize",
+		TokenURL: "https://auth.tesla.com/oauth2/v3/token",
+	},
+	Scopes: []string{"openid", "email", "offline_access"},
 }
 
-// BaseURL is the base URL of the standard Tesla API.
-const BaseURL = "https://owner-api.teslamotors.com/api/1"
-
 // Client provides the client and associated elements for interacting with the Tesla API.
 type Client struct {
 	BaseURL      string
```

Comment:
Do we want to expose the BaseURL always?

Reply:
I un-exported BaseURL and StreamingURL.